src/openrct2/object/RideObject.cpp

```diff
@@ -229,16 +229,28 @@ void RideObject::Load()
             uint32_t imageIndex = baseImageId;
             carEntry->base_image_id = baseImageId;
 
-            for (uint8_t spriteGroup = 0; spriteGroup < EnumValue(SpriteGroupType::Count); spriteGroup++)
+            switch (carEntry->PaintStyle)
             {
-                if (carEntry->SpriteGroups[spriteGroup].Enabled())
-                {
-                    carEntry->SpriteGroups[spriteGroup].imageId = imageIndex;
-                    const auto spriteCount = carEntry->base_num_frames
-                        * carEntry->NumRotationSprites(static_cast<SpriteGroupType>(spriteGroup))
-                        * SpriteGroupMultiplier[spriteGroup];
-                    imageIndex += spriteCount;
-                }
+                case VEHICLE_VISUAL_RIVER_RAPIDS:
+                case VEHICLE_VISUAL_VIRGINIA_REEL:
+                    carEntry->SpriteGroups[EnumValue(SpriteGroupType::SlopeFlat)] = { baseImageId, SpritePrecision::Sprites16 };
+                    carEntry->SpriteGroups[EnumValue(SpriteGroupType::Slopes12)] = { baseImageId + 8,
+                                                                                     SpritePrecision::Sprites4 };
+                    carEntry->SpriteGroups[EnumValue(SpriteGroupType::Slopes25)] = { baseImageId + 40,
+                                                                                     SpritePrecision::Sprites4 };
```

Comment:
Where do these magic offsets (8 and 40) come from? Did they exist before, but were removed?

Reply:
They come from the number of sprites involved in these sprite groups: 8 flat sprites, 32 slope transition sprites, 32 slope sprites. These values are unused by the paint code but used by the limit-track-elements code that was added recently.

Reply:
Figured something like that was the case here - are there no constants to use here instead of magic numbers?

Reply:
As far as I know, no constants exist for spinning rides - it's all done through magic numbers in VirgniaReel.cpp and RiverRapids.cpp.

Reply:
Can you introduce `constexpr` for those values? It’s OK if they’re just static to the RideObject.cpp file.

Reply:
If there are too many, we can just merge this PR and leave that for another PR. Just figured this is a good opportunity to prevent introducing more magic numbers to the codebase.